internal/cmd/commands/database/init.go

```diff
@@ -321,178 +325,176 @@ func (c *InitCommand) Run(args []string) (retCode int) {
 		}()
 	}
 
-	if c.flagSkipInitialLoginRoleCreation {
-		return base.CommandSuccess
-	}
-
-	role, err := c.CreateInitialLoginRole(c.Context)
-	if err != nil {
-		c.UI.Error(fmt.Errorf("Error creating initial global-scoped login role: %w", err).Error())
-		return base.CommandCliError
-	}
-
-	roleInfo := &RoleInfo{
-		RoleId: role.PublicId,
-		Name:   role.Name,
-	}
-	switch base.Format(c.UI) {
-	case "table":
-		c.UI.Output(generateInitialLoginRoleTableOutput(roleInfo))
-	case "json":
-		jsonMap["login_role"] = roleInfo
-	}
-
-	if c.flagSkipInitialAuthenticatedUserRoleCreation {
-		return base.CommandSuccess
-	}
-
-	role, err = c.CreateInitialAuthenticatedUserRole(c.Context)
-	if err != nil {
-		c.UI.Error(fmt.Errorf("Error creating initial global-scoped authenticated user role: %w", err).Error())
-		return base.CommandCliError
-	}
-
-	roleInfo = &RoleInfo{
-		RoleId: role.PublicId,
-		Name:   role.Name,
-	}
-	switch base.Format(c.UI) {
-	case "table":
-		c.UI.Output(generateInitialAuthenticatedUserRoleOutput(roleInfo))
-	case "json":
-		jsonMap["authenticated_user_role"] = roleInfo
-	}
-
-	if c.flagSkipAuthMethodCreation {
-		return base.CommandSuccess
-	}
-
-	// Use an easy name, at least
-	c.DevLoginName = "admin"
-	am, user, err := c.CreateInitialPasswordAuthMethod(c.Context)
-	if err != nil {
-		c.UI.Error(fmt.Errorf("Error creating initial auth method and user: %w", err).Error())
-		return base.CommandCliError
-	}
-
-	authMethodInfo := &AuthInfo{
-		AuthMethodId:   c.DevPasswordAuthMethodId,
-		AuthMethodName: am.Name,
-		LoginName:      c.DevLoginName,
-		Password:       c.DevPassword,
-		ScopeId:        scope.Global.String(),
-		UserId:         user.PublicId,
-		UserName:       user.Name,
-	}
-	switch base.Format(c.UI) {
-	case "table":
-		c.UI.Output(generateInitialAuthTableOutput(authMethodInfo))
-	case "json":
-		jsonMap["auth_method"] = authMethodInfo
+	if c.InitFlags.SkipInitialLoginRoleCreation() {
+		c.UI.Info("Skipping creation of initial login role.")
```

Comment:
I wonder if it would be nice to include a little summary for the user to explain why this was skipped. If each of these helper methods returned a string explaining _why_ instead of just a bool, we could print that here.

```suggestion
	if shouldSkip, reason := c.InitFlags.SkipInitialLoginRoleCreation(); shouldSkip{
		c.UI.Infof("Skipping creation of initial login role: %s.", reason)
```

Could print

```
Skipping creation of initial login role: flag `-skip-initial-login-role` was set
```

What do you think?

Reply:
Love the idea!

Reply:
Alright updated some of the logic for this, let me know what ya think!